python/fledge/services/core/api/update.py

```diff
@@ -107,3 +110,69 @@ async def update_package(request):
         raise web.HTTPInternalServerError(reason=msg, body=json.dumps({"message": msg}))
     else:
         return web.json_response({"status": "Running", "message": status_message})
+
+
+async def get_updates(request: web.Request) -> web.Response:
+    """
+        Gives list of packages for which updates are available.
+
+        Returns JSON Response
+         Sample Response
+         {
+            "updates": [
+            "fledge-south-sinusoid","fledge"
+            ]
+        }
+        Example
+         curl -sX GET http://localhost:8081/fledge/update |jq
+    """
+    _platform = platform.platform()
+    if "centos" in _platform or "redhat" in _platform:
+        update_cmd = "sudo yum update"
+        packages_check_cmd = "yum list updates | grep \^fledge"
+    else:
+        update_cmd = "sudo apt update"
+        packages_check_cmd = "apt list --upgradable | grep \^fledge"
+
+    update_process = await asyncio.create_subprocess_shell(update_cmd,
+                                                           stdout=asyncio.subprocess.PIPE,
+                                                           stderr=asyncio.subprocess.PIPE)
+
+    _, stderr = await update_process.communicate()
+    if update_process.returncode != 0:
+        msg = "Could not run {} due to {}".format(update_cmd, stderr.decode('utf-8'))
+        raise web.HTTPInternalServerError(reason=msg, body=json.dumps({"message": msg}))
+
+    installed_packages_process = await asyncio.create_subprocess_shell(packages_check_cmd,
+                                                                       stdout=asyncio.subprocess.PIPE,
+                                                                       stderr=asyncio.subprocess.PIPE)
+
+    stdout, stderr = await installed_packages_process.communicate()
+    if installed_packages_process.returncode != 0:
+        _logger.info("Nothing to upgrade at the moment. {}".format(stderr))
```

Comment:
We should add a comment why are we considering != 0 exit code as INFO.

Probably we should put stderr in a new log statement after `Nothing to upgrade at the moment, for the configured repository.`